src/protocols/secure_channel/CASESession.cpp

```diff
@@ -875,25 +905,26 @@ CHIP_ERROR CASESession::SendSigmaR3()
         TLV::TLVType outerContainerType = TLV::kTLVType_NotSpecified;
 
         tlvWriter.Init(msg_R3_Encrypted.Get(), msg_r3_encrypted_len);
-        err = tlvWriter.StartContainer(TLV::AnonymousTag, TLV::kTLVType_Structure, outerContainerType);
-        SuccessOrExit(err);
-        err = tlvWriter.PutBytes(CASETLVTag::kNOC, mOpCredSet->GetDevOpCred(mTrustedRootId),
-                                 mOpCredSet->GetDevOpCredLen(mTrustedRootId));
-        SuccessOrExit(err);
-        err = tlvWriter.PutBytes(CASETLVTag::kSignature, sigmaR3Signature, static_cast<uint32_t>(sigmaR3Signature.Length()));
-        SuccessOrExit(err);
-        err = tlvWriter.EndContainer(outerContainerType);
-        SuccessOrExit(err);
-        err = tlvWriter.Finalize();
-        SuccessOrExit(err);
+        SuccessOrExit(err = tlvWriter.StartContainer(TLV::AnonymousTag, TLV::kTLVType_Structure, outerContainerType));
+        SuccessOrExit(err = tlvWriter.PutBytes(CASETLVTag::kNOC, mOpCredSet->GetDevOpCred(mTrustedRootId),
```

Comment:
Is GetDevOpCred already a cert array? I think @bzbarsky-apple was talking about maybe splitting the NOC+ICAC cert arrays into separate fields everywhere to simplify things. No need to do in this PR, but something where a conversation could be had.

Reply:
Agreed, could be something to look into for the future.